src/models.rs

```diff
@@ -0,0 +1,38 @@
+use near_sdk::serde::{Serialize, Deserialize};
+
+use crate::*;
+
+/// Information about a specific public key. Should be returned in the `get_key_information` view method.
+/// Part of the linkdrop NEP
+#[derive(Serialize)]
+#[serde(crate = "near_sdk::serde")]
+pub struct KeyInfo {
+    /// yoctoNEAR$ amount that will be sent to the claiming account (either new or existing)
+    /// when the key is successfully used.
+    pub balance: U128,
+}
+
+
+#[derive(Serialize, Deserialize)]
+#[serde(crate = "near_sdk::serde")]
+/// Information about any limited access keys that are being added to the account as part of `create_account_advanced`.
+pub struct LimitedAccessKey {
+    /// The public key of the limited access key.
+    pub public_key: PublicKey,
+    /// The amount of yoctoNEAR$ that can be spent on Gas by this key.
+    pub allowance: U128,
```

Comment:
What about the case for unlimited allowance? Usually, defined as `Option<U128>` in https://docs.rs/near-primitives/0.15.0/near_primitives/account/struct.FunctionCallPermission.html

You gotta be careful about this case in particular, since `0` value means unlimited when it comes to promises. I'd suggest adding a check and erroring out when the user supplies a `0` for the Allowance, whereas `None` should specify unlimited.

Actually, there should have been warnings when you called into `add_access_key`. `add_access_key_allowance` is better as that tells you that you need to use a new type called `Allowance` which checks against this case

Reply:
I was assuming that if they wanted unlimited allowance, they would just manually pass in `0`. This is safer than having an `Option` since if they forget to pass in allowance, it defaults to unlimited.

Reply:
I meant better to make your user supply an `Allowance` type than make them specify `0`. The `0` can be quite sketchy because it actually can mean different things for allowance in different contexts. Like for promises, it would mean unlimited, but on a RPC call, it literally means zero allowance

Reply:
Oh I see - where can I find that data type? It doesn't look like it's in the Rust SDK

Reply:
ahh, it hasn't been released. That's probably why you didn't see warnings for it:
https://github.com/near/near-sdk-rs/blob/89f8c2d776c8d00feb45239fdb15725527f1894b/near-sdk/src/promise.rs#L13-L17